src/Database/Database.php

```diff
@@ -265,6 +281,40 @@ function (?string $value) {
         );
     }
 
+    /**
+     * Add listener to events
+     *
+     * @param string $event
+     * @param callable $callback
+     * @return self
+     */
+    public function on(string $event, callable $callback): self
+    {
+        if(!isset($this->listeners[$event])) {
+            $this->listeners[$event] = [];
+        }
+        $this->listeners[$event][] = $callback;
+        return $this;
+    }
+
+    /**
+     * Trigger callback for events
+     *
+     * @param string $event
+     * @param array|null $args
+     * @return void
+     */
+    protected function trigger(string $event, mixed $args = null): void
+    {
+        foreach ($this->listeners['*'] as $callback) {
```

Comment:
```suggestion
        foreach ($this->listeners['*'] as $callback) {
```

We can have a constant for all events as well

Reply:
updated